angular-portfolio/src/app/lifecycle-practice/overview-parent/overview-parent.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component } from '@angular/core';
+import { LoggerService } from '../logger.service';
+
+@Component({
+    selector: 'app-overview-parent',
+    templateUrl: './overview-parent.component.html',
+    styleUrls: ['./overview-parent.component.scss'],
+})
+export class OverviewParentComponent {
+    hookLog: string[];
+    hookColorArray: string[] = [];
+    name: string;
+    hasChild = false;
+    private possibleNames = ['Billy', 'Matt', 'Nicholas', 'Sarah', 'Sofia'];
+    private nextName: string;
+    private logger: LoggerService;
+
+    constructor(logger: LoggerService) {
```

Comment:
in the constructor arguments, just write "(private readonly logger: LoggerService)"

Then you don't need to initialize "this.logger = logger" or declare the property.

In other words, you can declare properties in the constructor arguments.